projects/fsastorefrontlib/src/cms-components/agent/agent-sync-pilot/agent-sync-pilot.component.ts

```diff
@@ -0,0 +1,67 @@
+import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
+import { Observable } from 'rxjs';
+import { CmsService, User, WindowRef } from '@spartacus/core';
+import {
+  CmsComponentData,
+  IconConfig,
+  ModalService,
+} from '@spartacus/storefront';
+import { UserAccountFacade } from '@spartacus/user/account/root';
+import { Service } from '@syncpilot/bpool-guest-lib';
+import { SyncPilotConnectionComponent } from '../../sync-pilot/sync-pilot-connection.component';
+import { CMSConnectionComponent } from '../../../occ/occ-models/cms-component.models';
+import { tap } from 'rxjs/operators';
+import { SyncPilotGender } from '../../../occ/occ-models/occ.models';
+import { ICON_TYPE } from '../../../core/icon-config/icon-config';
+
+@Component({
+  selector: 'cx-fs-agent-sync-pilot',
+  templateUrl: './agent-sync-pilot.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class AgentSyncPilotComponent extends SyncPilotConnectionComponent {
+  @Input() agent;
+  @Input() set type(type: ICON_TYPE) {
+    this.agentIcon = this.iconConfig.icon.symbols[type];
+  }
+  agentIcon: string = this.iconConfig.icon.symbols[ICON_TYPE.PHONE];
+  component$: Observable<any> = this.cmsService.getComponentData(
+    'AgentSyncPilotComponent'
```

Comment:
component$: Observable<any> = this.componentData.data$;

It's better to dynamically fetch data from the backend. no need to be hardcoded to AgentSyncPilotComponent

Reply:
Since `AgentSyncPilotComponent ` is not mapped as cms component I need to use `cmsService`.